app/main.py

```diff
@@ -2,5 +2,15 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_list = {}
```

Comment:
The `cache_list` dictionary is used to store cached results. However, using mutable types (like lists) as keys in a dictionary will raise a `TypeError`. Ensure that the arguments used as keys are immutable types, such as tuples.